Arm move group

Rubyboat1207 opened the pull request.

5431-programming committed:
PresetPosition stores presets fr

also rename JumpToGoalPosition to ArmToGoal (more concise)
get rid of WristAngle as PresetPosition now includes wrist angle
KinematicsSolver returns pair of angles, not presetposition

5431-programming committed:
Update deps

5431-programming committed:
Silence disconnected joysticks in sim

Rubyboat1207 committed:
Presets and command groups

Rubyboat1207 committed:
fixed fr commit

Rubyboat1207 committed:
Merge branch 'master' of https://github.com/frc5431/RobotCode2023 into arm-move-group

democat3457 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -126,34 +127,73 @@ private void configureBindings() {
         ));
 
-        operator.povRight().onTrue(new ArmToGoalCommand( // High node
+        operator.povRight().onTrue(new WristAngleCommand( // High node
```
why is this command format different than the previous

democat3457 commented on src/main/java/frc/robot/commands/WristAngleCommand.java:
```diff
@@ -0,0 +1,26 @@
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Arm.ArmComponent;
+
+public class WristAngleCommand extends CommandBase {
```
can you use the new parameters to set wrist angle on ArmToGoalCommand ctor pls

democat3457 commented on src/main/java/frc/robot/commands/WristAngleCommand.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Override
+    public boolean isFinished() {
+        return wrist.atSetpoint();
```
also add this to the isFinished of ArmToGoalCommand if it's not already in there

democat3457 commented on src/main/java/frc/robot/commands/WristOpenCommand.java:
```diff
@@ -0,0 +1,24 @@
+import frc.robot.subsystems.Arm;
+import frc.robot.subsystems.Manipulator;
+
+public class WristOpenCommand extends CommandBase {
```
look into InstantCommand, and make into a Command factory method of Manipulator - see 2022's subsystems for examples

democat3457 replied:
I'll also commit what I'm looking for this

democat3457 commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -61,8 +61,8 @@ public class Arm extends SubsystemBase {
 
-    private final double SETPOINT_POSITION_TOLERANCE = Units.degreesToRadians(1);
-    private final double SETPOINT_VELOCITY_TOLERANCE = Units.degreesToRadians(5);
+    private final double SETPOINT_POSITION_TOLERANCE = 2.5;
```
this is in radians -.- where did the degreesToRadians go?

Rubyboat1207 replied:
supposedly, however this made things have intended behavior

democat3457 replied:
fine... ill look at it later

democat3457 commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -61,8 +61,8 @@ public class Arm extends SubsystemBase {
-    private final double SETPOINT_POSITION_TOLERANCE = Units.degreesToRadians(1);
-    private final double SETPOINT_VELOCITY_TOLERANCE = Units.degreesToRadians(5);
+    private final double SETPOINT_POSITION_TOLERANCE = 2.5;
+    private final double SETPOINT_VELOCITY_TOLERANCE = 5;
```
where did degreesToRadians go?

democat3457 commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -237,11 +237,13 @@ public void periodic() {
         SmartDashboard.putBoolean("elbow atSetpoint", innerComponent.atSetpoint());
         SmartDashboard.putBoolean("wrist atSetpoint", wristComponent.atSetpoint());
 
+        SmartDashboard.putNumber("ingebinge", Math.abs(wristComponent.absoluteEncoder.getPosition() - wristComponent.setpoint));
```
rename the entry name

democat3457 commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -334,6 +337,19 @@ public ArmComponent(CANSparkMax motor, CANSparkMax follow, MotionMagic pidConsta
+        }
+
+        public void UpdateDash() {
+            if(name.isEmpty()) {
```
do you mean !isEmpty?

democat3457 commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -334,6 +337,19 @@ public ArmComponent(CANSparkMax motor, CANSparkMax follow, MotionMagic pidConsta
+            this.name = name;
+        }
+
+        public void UpdateDash() {
```
use lowerCamelCase

democat3457 replied:
also, where is this method even used

Rubyboat1207 replied:
later, i have plans. this was made after the pr

democat3457 reviewed (changes requested):
Looks good, but needs some work.

Rubyboat1207 committed:
Use colins code finally. fix other stuff

democat3457 commented on src/main/java/frc/robot/commands/ArmToGoalCommand.java:
```diff
@@ -55,6 +61,9 @@ public void initialize() {
 
     @Override
     public void execute() {
+        if((flags & IGNORE_POSITION) == IGNORE_POSITION) {
```
I would actually argue for the reinstatement of WristAngleCommand - but in a very specific way. I'll commit what I'm thinking of, and then this flag can be removed. The reason I don't like this flag is because it completely nullifies a part of the command, and at this point you may as well write a more concise command.

democat3457 commented on src/main/java/frc/robot/commands/ArmToGoalCommand.java:
```diff
@@ -26,7 +28,11 @@ public class ArmToGoalCommand extends CommandBase {
-        this.wristDegrees = presetPosition.getWrist();
+
+        //Untested, but i think it'll work
+        if((flags & IGNORE_ROTATION) == IGNORE_ROTATION) {
```
a) this is inverted
b) if you wanted it to ignore rotation, just use the constructor below that accepts a Translation2d instead of the PresetPosition - you can use `PresetPosition#getGoal` if you have the PresetPosition instead.

democat3457 commented on src/main/java/frc/robot/commands/ArmToGoalCommand.java:
```diff
@@ -63,17 +72,17 @@ public void execute() {
         double distanceFromGoal = pos.minus(goalPosition).getNorm();
-        
-        return Math.abs(distanceFromGoal) <= DISTANCE_TOLERANCE; // distance should already be absolute, but doesn't hurt to check
+        return (Math.abs(distanceFromGoal) <= DISTANCE_TOLERANCE || ((flags & IGNORE_POSITION) == 0)) && arm.getWrist().atSetpoint(); // distance should already be absolute, but doesn't hurt to check
```
Only confirm is wrist is atSetpoint if the setpoint above is NaN (otherwise we'll be checking for wrist even if we didn't set it, which is outside the scope of the command's function)

democat3457 committed:
Remove extraneous flags and move functionality

- Remove WristOpenCommand and move the functionality to Manipulator: see runOnce inline command factory (runOnce also creates a command that already requires the containing SubsystemBse)
- Remove the getStateFromBool and setState, as they dont mess with isOpen and are no longer needed if WristOpen is removed
- Make Manipulator and Deadwheels extend SubsystemBase to enable the SubsystemBase command factories
- Remove RunEndCommand as that already exists in Commands factories
- Remove the new flags for ArmToGoal:
  - IGNORE_ROTATION can be removed, as the second constructor exists if you want to avoid setting wrist
  - IGNORE_POSITION can be removed as it takes out most of ArmToGoal's original functionality - see below
- Add setDegreesCommand command factory to arm components- this will set the setpoint, then wait until component is at setpoint before continuing. (for the entire duration of the command, the whole Arm will be required, so only one of these commands can be called at a time!)
- Fix lowerCamelCase for UpdateDash
